llvm-15.0.3/llvm-crash-analyzer/include/Target/CATargetInfo.h

```diff
@@ -53,34 +56,47 @@ class CATargetInfo {
 
 public:
   CATargetInfo() {}
-  virtual ~CATargetInfo() { RegMap.clear(); InstAddrs.clear(); }
+  virtual ~CATargetInfo() {
+    RegMap.clear();
+    InstAddrs.clear();
+  }
 
   // Get register index in the RegMap.
   virtual Optional<unsigned> getID(std::string RegName) const = 0;
 
+  // Get register unsigned (MCRegister) from the RegMap.
+  virtual Optional<unsigned> getRegister(std::string RegName,
+                                         const MachineInstr *MI) const = 0;
+
   virtual unsigned getRegSize(std::string RegName) const = 0;
 
   // Get RegAliasTuple from the RegMap with selected Id.
   RegAliasTuple &getRegMap(unsigned Id) const {
     return const_cast<RegAliasTuple &>(RegMap.at(Id));
   }
 
+  // Get RegAliasTuple from the RegMap with selected Id.
+  std::unordered_map<unsigned, RegAliasTuple> getWholeRegMap() const {
```

Comment:
better return a reference to map instead of copying the entire map; no?

Reply:
Definitely, I missed this, thanks!